core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -447,6 +448,51 @@ private function addMailerNode(): NodeDefinition
         ;
     }
 
+    private function addBackendNode(): NodeDefinition
+    {
+        return (new TreeBuilder('backend'))
+            ->getRootNode()
+            ->addDefaultsIfNotSet()
+            ->children()
+                ->arrayNode('attributes')
+                    ->info('Adds HTML data attributes to the <body> tag in the back end.')
+                    ->example(['app-name' => 'My App', 'app-version' => '1.2.3'])
+                    ->validate()
+                    ->always(
+                        static function (array $attributes): array {
+                            foreach (array_keys($attributes) as $name) {
+                                if (preg_match('/[^a-z0-9\-\.:_]/', (string) $name)) {
```

Comment:
Unit tests for that case would be great and I think you can just abbreviate the error message and use the regex itself as explanation. People should be able to read the regex.

Reply:
I only orientated myself on this error message:
https://github.com/contao/contao/blob/f53d8cf6cb5334a09463fd7c9af27a0fe2a90341/core-bundle/src/DependencyInjection/Configuration.php#L215

Reply:
What about something like `The attribute name "%s" must be a valid HTML attribute name.`?

The [HTML spec](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2) is pretty loose on this. btw.

Reply:
Okay good idea - I changed the error message! Well, I think my regex should match the common usages, shouldn't it?

Reply:
I wouldn't do it as loose as the spec, yes. It's not even sure most browsers behave like this.